src/permissions/plugins/github/index.ts

```diff
@@ -0,0 +1,130 @@
+import { MessageBuilder } from '../../../MessageBuilder.js';
+import { RepositoryConfig, TeamConfig } from '../../types.js';
+import { Plugin } from '../Plugin.js';
+import { GITHUB_DEFAULT_BRANCH, GITHUB_DEFAULT_ENVIRONMENT } from '../../../constants.js';
+import { IS_DRY_RUN } from '../../../helpers.js';
+import { getOctokit } from '../../../octokit.js';
+import chalk from 'chalk';
+
+class GitHubPlugin implements Plugin {
+  handleRepo = async (
+    repo: RepositoryConfig,
+    teams: TeamConfig[],
+    org: string,
+    builder: MessageBuilder,
+  ) => {
+    const { properties } = repo;
+    if (!properties) return;
+
+    if (properties.type === 'ecosystem-npm-package') {
+      await this.ensureNpmEnvironment(repo, org, builder);
+    }
+  };
+
+  private ensureNpmEnvironment = async (
+    repo: RepositoryConfig,
+    org: string,
+    builder: MessageBuilder,
+  ) => {
+    const octokit = await getOctokit(org);
+
+    try {
+      // Check if the environment already exists
+      let environmentExists = false;
+      try {
+        await octokit.repos.getEnvironment({
```

Comment:
I think we also need to check `environment.deployment_branch_policy.custom_branch_policies === true` here?